vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution_unblinded.cc

```diff
@@ -198,9 +200,73 @@ void Unblinded::PrepareTokens(
     return;
   }
 
-  // TODO(https://github.com/brave/brave-browser/issues/8887):
-  // we should reserve this tokens and add step STEP_RESERVE
-  PreparePublishers(token_list, std::move(contribution), types, callback);
+  const std::string contribution_id = contribution->contribution_id;
+  const std::string contribution_string =
+      braveledger_bind_util::FromContributionToString(std::move(contribution));
+
+  auto save_callback = std::bind(&Unblinded::ReserveStepSaved,
+      this,
+      _1,
+      token_list,
+      contribution_id,
+      contribution_string,
+      types,
+      callback);
+
+  ledger_->UpdateContributionInfoStep(
```

Comment:
we should first reserve tokens and then change status. because if status is saved and tokens are not you are stuck